posts/test_post_serving.py

```diff
@@ -0,0 +1,106 @@
+"""Unit tests for posts serving."""
+
+# Authors: mukobi
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import unittest
+import mongomock
+import app
+
+FAKE_POSTS = [
+    {
+        "event_id": "picnic",
+        "author_id": "mukobi",
+        "created_at": "2017-10-06T00:00:00+00:00",
+        "text": "This is the description of a picnic post.",
+        "images": [
+            "https://upload.wikimedia.org/wikipedia/commons/6/66/"
+            "An_up-close_picture_of_a_curious_male_domestic_short"
+            "hair_tabby_cat.jpg",
+            "http://4.bp.blogspot.com/-w8U75TCuhgU/Tzw8TmaclvI/AA"
+            "AAAAAABJ0/6fMMcRLAceM/s1600/Rabbit3.jpg"
+        ]
+    },
+    {
+        "event_id": "aquarium",
+        "author_id": "mukobi",
+        "created_at": "2011-11-11T00:00:00+00:00",
+        "text": "This is the description of an aquarium post.",
+        "images": [
+            "https://upload.wikimedia.org/wikipedia/commons/0/02/"
+            "Sea_Otter_%28Enhydra_lutris%29_%2825169790524%29_crop.jpg"
+        ]
+    },
+    {
+        "event_id": "aquarium",
+        "author_id": "john_wick",
+        "created_at": "2014-10-24T00:00:00+00:00",
+        "text": "This is another aquarium post.",
+        "images": []
+    }
+]
+
+
+class TestPostDBGetting(unittest.TestCase):
+    """Test app.find_posts_in_db()."""
+
+    def setUp(self):
+        """Seed mock db."""
+        self.collection = mongomock.MongoClient().db.collection
+        self.collection.insert_many(FAKE_POSTS)
+
+    def test_find_all_posts(self):
+        """Find all fake posts."""
+        found = app.find_posts_in_db(self.collection)
+        self.assertEqual(found, FAKE_POSTS)
+
+    def test_find_by_post_id(self):
+        """Search for 1 post by post ID."""
+        post_id = FAKE_POSTS[0]["_id"]
```

Comment:
I don't see where an "_id" attribute is defined in FAKE_POSTS, where does it come from?

Reply:
I just found this out today: `collection.insert_many(document)` will actually add a `"_id"` attribute to `document` to match the ID that is generated in the MongoDB collection if you didn't already add a `"_id"` to `document`, which is pretty cool. I found this out when a test comparing the fake object to the one in the db said they were the same which I was not expecting. I've gone ahead an added a comment explaining this https://github.com/knative-portability/large-events/pull/24/commits/de1938ec530ea005e406717d092602c296daf6ea